### PR TITLE
Designer action panel line refactoring

### PR DESCRIPTION
**Pull request by halgab**
Depends on #9367. Only the last commit is not in the quoted PR and should be reviewed here.

As dicussed [here](https://github.com/dotnet/winforms/pull/9367#discussion_r1251266055), we're attempting a refactoring to limit the use of the `!` operator when annotating `DesignerActionPanel.Line` objects.

The idea is to stop creating `Line` objects early without initialising them and start creating then only when we are actually going to use them. To do so we change the `LineInfo` type to be a factory type.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9422)

### CODE REVIEW

**Comment by halgab** [comment]
@dreddy-work

**Comment by dreddy-work** [comment]
This change has to wait until branch open for 9.0.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.SeparatorLine.cs [hunk 1]
```diff
@@ -10,26 +10,14 @@ internal sealed partial class DesignerActionPanel
...
-        {
-            get => string.Empty;
-        }
-
-        public bool IsSubSeparator => _isSubSeparator;
-
-        protected override void AddControls(List<Control> controls)
-        {
-        }
+        public bool IsSubSeparator { get; }
```
Comment (halgab) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.SeparatorLine.cs:20]: Reviewing this code again, this property looks totally unused... Not sure if it's expected

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs [hunk 2]
```diff
@@ -62,7 +58,17 @@ protected override void OnPropertyTaskItemUpdated(ToolTip toolTip, ref int curre
 
         protected override void OnValueChanged()
         {
-            _checkBox!.Checked = (bool)Value!;
+            _checkBox.Checked = (bool)Value!;
+        }
+
+        public sealed class Info(DesignerActionList list, DesignerActionItem item) : LineInfo(list, item)
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs]: Where is this used?

**Comment by dreddy-work** [comment]
@halgab, refactor looks good to me. To make review process simple, I would try push this refactoring first and then work on nullability. Can you make a PR just with refactoring with reasoning behind for review?

**Comment by halgab** [comment]
Closing this in favor of #10035 and #9367 (or its successor)

**Pull request closed without merging**